### PR TITLE
Feat/load from zip

### PR DESCRIPTION
**Pull request by ComicallyNormal**
Hi. I am working on a project downstream of this sdk, that would need to utilize loading from an xmind file and interpreting data from it. Specifically, we will be using trees that will be compared against a valid schema, using custom markers to validate that nodes are at the level in the tree that they are supposed to be at. 

I have updated the feature branch with code from master, and implemented basic loading functionality for this project. I would like some feedback, especially about any problems that you guys have run into in the past in attempting to implement this functionality. 

Please do not merge this immediately, there are some architectural problems ( the biggest of which is that my loading functions are in the Topic class) that I would like to sort out before considering this ready to merge.

Thank you for your time, and apologies if I am approaching this process wrong, this is my first stab really at open source.

### CODE REVIEW

**Comment by danielsss** [comment]
> Hi. I am working on a project downstream of this sdk, that would need to utilize loading from an xmind file and interpreting data from it. Specifically, we will be using trees that will be compared against a valid schema, using custom markers to validate that nodes are at the level in the tree that they are supposed to be at.
> 
> I have updated the feature branch with code from master, and implemented basic loading functionality for this project. I would like some feedback, especially about any problems that you guys have run into in the past in attempting to implement this functionality.
> 
> Please do not merge this immediately, there are some architectural problems ( the biggest of which is that my loading functions are in the Topic class) that I would like to sort out before considering this ready to merge.
> 
> Thank you for your time, and apologies if I am approaching this process wrong, this is my first stab really at open source.

@ComicallyNormal Yes, you're right that we've tried to implement the feature before and also, there is an issue(#20) which explains why we didn't do that.

To be honest, the core model has been refactored internally, but we didn't make it open-source yet. I'd like to open an `extensional branch` if you really want this feature.

You also need to maintain or contribute this branch by yourself.

**Comment by ComicallyNormal** [comment]
> @ComicallyNormal Yes, you're right that we've tried to implement the feature before and also, there is an issue(#20) which explains why we didn't do that.
> 
Is attachment support the main issue here then? is that due to file references in a .xmind archive that are hard to parse, or some other reason? 

> To be honest, the core model has been refactored internally, but we didn't make it open-source yet. I'd like to open an `extensional branch` if you really want this feature.

I'm assuming that I will continue to work on the forked repo and submit MRs into the extensional branch then?

> You also need to maintain or contribute this branch by yourself.

This is not a problem, just wanted to make you aware of the work I was doing to avoid duplicating the implementatation.

**Comment by danielsss** [comment]
> > @ComicallyNormal Yes, you're right that we've tried to implement the feature before and also, there is an issue(#20) which explains why we didn't do that.
> 
> Is attachment support the main issue here then? is that due to file references in a .xmind archive that are hard to parse, or some other reason?
> 
> > To be honest, the core model has been refactored internally, but we didn't make it open-source yet. I'd like to open an `extensional branch` if you really want this feature.
> 
> I'm assuming that I will continue to work on the forked repo and submit MRs into the extensional branch then?
> 
> > You also need to maintain or contribute this branch by yourself.
> 
> This is not a problem, just wanted to make you aware of the work I was doing to avoid duplicating the implementatation.


1. It isn't only about the attachments. Basically, I need to implement all the features as much as our UI clients, because our client has more features than this SDK(it means that they might load a file containing features that have not been implemented here).

2. We can easily keep up-to-date with the master branch and release an individual package for you and others who want this feature. Anyway, It's up to you.

Thanks for letting me know what you are doing. Great Job.
